.github/bin/run-clang-tidy.sh

```diff
@@ -128,17 +129,35 @@ if [ -s ${FILES_TO_PROCESS} ]; then
   echo ::group::Summary
   sed 's|\(.*\)\(\[[a-zA-Z.-]*\]$\)|\2|p;d' < ${TIDY_OUT} | sort | uniq -c | sort -rn
   echo "::endgroup::"
-
-  if [ -s ${TIDY_OUT} ]; then
-    echo "::error::There were clang-tidy warnings. Please fix"
-    exit 1
-  fi
 else
   echo "Skipping clang-tidy run: nothing to do"
 fi
 
-# No complaints. We can cache this list now as baseline for next time.
-cp ${CLANG_TIDY_SEEN_CACHE}.new ${CLANG_TIDY_SEEN_CACHE}
 
-echo "No clang-tidy complaints.😎"
-exit 0
+if [ -s ${TIDY_OUT} ]; then
+  echo "::error::There were clang-tidy warnings. Please fix"
+  EXIT_CODE=1
+  # Assemble a sed expression that matches all files that were mentioned
+  # in the clang-tidy output.
+  # Use that to filter the file list cache to leave out files with failures,
+  # but keep the ones that had no issues.
+  # That way, only files that had issues will have to be processed in
+  # subsequent clang-tidy runs.
+  EXCLUDE_EXPRESSION=$(awk -F: '
+        /^[a-zA-Z_./]*:/ { seen[$1]=1; }
+        END {
+          is_first=1;
+          for (k in seen) {
+            printf("%s%s", is_first ? "/" : "\\|", gensub(/\//, "\\\\/", "g", k));
+            is_first=0;
+          }
+          printf("/d");
+        }' < ${TIDY_OUT})
+  sed "${EXCLUDE_EXPRESSION}" < ${CLANG_TIDY_SEEN_CACHE}.new > ${CLANG_TIDY_SEEN_CACHE}
```

Comment:
`sed -e` (to be explicit)

Alternatively, you could send the output of `awk` to a file and use `sed -f` on that.  Might run less risk of hitting string length limits in the shell.

Reply:
Good idea. Done.